swarm/network/stream/peer.go

```diff
@@ -420,3 +423,156 @@ func (p *Peer) close() {
 		s.Close()
 	}
 }
+
+// runUpdateSyncing is a long running function that creates the initial
+// syncing subscriptions to the peer and waits for neighbourhood depth change
+// to create new ones or quit existing ones based on the new neighbourhood depth
+// and if peer enters or leaves nearest neighbourhood by using
+// syncSubscriptionsDiff and updateSyncSubscriptions functions.
+func (p *Peer) runUpdateSyncing() {
+	timer := time.NewTimer(p.streamer.syncUpdateDelay)
+	defer timer.Stop()
+
+	select {
+	case <-timer.C:
+	case <-p.streamer.quit:
+		return
+	}
+
+	kad := p.streamer.delivery.kad
+	po := chunk.Proximity(network.NewAddr(p.Node()).Over(), kad.BaseAddr())
```

Comment:
Having looked at `network.NewAddr`, I doubt this is used everywhere in Swarm, because our bzz addrs are different to enode.IDs, whereas this function is defined as:
```
func NewAddr(node *enode.Node) *BzzAddr {
	return &BzzAddr{OAddr: node.ID().Bytes(), UAddr: []byte(node.String())}
}
```

Reply:
I might be wrong, but it seems to me that `network.NewAddr` is used only in tests and simulations, and in production Swarm the BzzAddr is generated differently. See `Swarm.Start` where we actually override the Underlay Address, and we keep the previously set Overlay Address:

```
	// update uaddr to correct enode
	newaddr := s.bzz.UpdateLocalAddr([]byte(srv.Self().String()))
	log.Info("Updated bzz local addr", "oaddr", fmt.Sprintf("%x", newaddr.OAddr), "uaddr", fmt.Sprintf("%s", newaddr.UAddr))
```

Reply:
`handleChunkDelivery` also appears to be calculating proximity incorrectly based on `enode.ID` and not based on its `BzzAddr`:

```
	case *ChunkDeliveryMsgRetrieval:
		msg = (*ChunkDeliveryMsg)(r)
		peerPO := chunk.Proximity(sp.ID().Bytes(), msg.Addr)
```

Reply:
Good find, thanks for the fix in your PR.